packages/x-date-pickers-pro/src/DateRangePickerDay/DateRangePickerDay.tsx

```diff
@@ -243,6 +239,7 @@ const DateRangePickerDayRaw = React.forwardRef(function DateRangePickerDay<TDate
       data-mui-test={shouldRenderHighlight ? 'DateRangeHighlight' : undefined}
       className={clsx(classes.root, className)}
       ownerState={ownerState}
+      sx={sx}
```

Comment:
As far as I understand, the `sx` prop should always be passed to the root because otherwise you can't use it for the root.

But that's a breaking change I did not anticipate

Reply:
@alexfauquette I'd be curious to have your opinion on this one

Reply:
@LukasTy I'm not sure to understand how you did the yellow customization.

It sounds like a breaking change and a bug fix at the same time, because with this `sx` prop I would expect to customize days with 

```js
sx={{
"& .MuiPickersDay-root" : { ... }
}}
```

Reply:
In theory, I think we should even spread the `other` to `DateRangePickerDayRoot` and only follow the date props to `DateRangePickerDayDay`, to have the conformance tests working.

Reply:
@alexfauquette I simply put following `sx` on the `Day` component in `renderDay` function.
```tsx
sx={{
  backgroundColor: day.isSame(dayjs()) ? 'red' : 'yellow'
}}
```
The same works fine for `DatePicker` and currently does work for `DateRangePicker`.
But in any case. Sure, we could go forward with this `bugfix` and just add it in migration docs. 🙈

Reply:
@flaviendelangle @alexfauquette What do you think about postponing this PR or even better, only this change of where the props are spread on this component for v6 to avoid potential regressions for users? 🤔

Reply:
We have to confirm on Monday but next weeks release should be a v6 alpha
So no problem postponing :+1: